spec/codecs/netflow_spec.rb

```diff
@@ -1 +1,208 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/codecs/netflow"
+
+describe LogStash::Codecs::Netflow do
+  subject do
+    LogStash::Codecs::Netflow.new.tap do |codec|
+      expect{codec.register}.not_to raise_error
+    end
+  end
+
+  let(:decode) do
+    [].tap do |events|
+      subject.decode(data){|event| events << event}
+    end
+  end
+
+  context "Netflow 5" do
+    let(:data) do
+      # this netflow raw data was produced with softflowd and captured with netcat
+      # softflowd -D -i eth0 -v 5 -t maxlife=1 -n 127.0.01:8765
+      # nc -k -4 -u -l 127.0.0.1 8765 > netflow5.dat
+      IO.read(File.join(File.dirname(__FILE__), "netflow5.dat"), :mode => "rb")
+    end
+
+    let(:json_events) do
+      events = []
+      events << <<-END
+        {
+          "@timestamp": "2015-05-02T18:38:08.280Z",
+          "netflow": {
+            "version": 5,
+            "flow_seq_num": 0,
+            "engine_type": 0,
+            "engine_id": 0,
+            "sampling_algorithm": 0,
+            "sampling_interval": 0,
+            "flow_records": 2,
+            "ipv4_src_addr": "10.0.2.2",
+            "ipv4_dst_addr": "10.0.2.15",
+            "ipv4_next_hop": "0.0.0.0",
+            "input_snmp": 0,
+            "output_snmp": 0,
+            "in_pkts": 5,
+            "in_bytes": 230,
+            "first_switched": "2015-06-21T11:40:52.280Z",
+            "last_switched": "2015-05-02T18:38:08.279Z",
+            "l4_src_port": 54435,
+            "l4_dst_port": 22,
+            "tcp_flags": 16,
+            "protocol": 6,
+            "src_tos": 0,
+            "src_as": 0,
+            "dst_as": 0,
+            "src_mask": 0,
+            "dst_mask": 0
+          },
+          "@version": "1"
+        }
+      END
+
+      events << <<-END
+        {
+          "@timestamp": "2015-05-02T18:38:08.280Z",
+          "netflow": {
+            "version": 5,
+            "flow_seq_num": 0,
+            "engine_type": 0,
+            "engine_id": 0,
+            "sampling_algorithm": 0,
+            "sampling_interval": 0,
+            "flow_records": 2,
+            "ipv4_src_addr": "10.0.2.15",
+            "ipv4_dst_addr": "10.0.2.2",
+            "ipv4_next_hop": "0.0.0.0",
+            "input_snmp": 0,
+            "output_snmp": 0,
+            "in_pkts": 4,
+            "in_bytes": 304,
+            "first_switched": "2015-06-21T11:40:52.280Z",
+            "last_switched": "2015-05-02T18:38:08.279Z",
+            "l4_src_port": 22,
+            "l4_dst_port": 54435,
+            "tcp_flags": 24,
+            "protocol": 6,
+            "src_tos": 0,
+            "src_as": 0,
+            "dst_as": 0,
+            "src_mask": 0,
+            "dst_mask": 0
+          },
+          "@version": "1"
+        }
+      END
+
+      events.map{|event| event.gsub(/\s+/, "")}
+    end
+
+    it "should decode raw data" do
```

Comment:
Hi, sorry if you took it personal, it was not my intention, I apologies for that. However I must say you're really checking more than one things in this tests for example:

* That it decodes all events.
* That it decodes more than one message at the same time.
* That decodes ints and also strings.
* That it decodes all the elements of in the pipeline of the codec.

Even if the result of your test are ok, then when you hit an error this test are harder to mantain, but is just my experience.

 Please feel free to ignore me.